internal/controller/services/gateway/gateway_auth_actions.go

```diff
@@ -36,6 +38,17 @@ const (
 	AuthModeNone            AuthMode = "None"
 )
 
+// make secret data into sha256 as hash.
+func calculateSecretHash(secretData map[string][]byte) string {
```

Comment:
Is it correct to keep this function inside gateway_support.go?

Reply:
not sure i understand the question
as "it should be in another package?" or "we should not have this fuction"?

Reply:
what you meant is:
- `gateway_auth_actions.go` should only host functions : `createKubeAuthProxyInfrastructure` `createEnvoyFilter`
- `gateway_controller_actions.go` only for funtions:  `createGatewayInfrastructure` `createDestinationRule` `syncGatewayConfigStatus`
all the rest functions should be in `gateway_support.go`?

Reply:
yes